src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunctionComparer.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.PowerFx.Core.Functions
+{
+    internal class TexlFunctionComparer : IEqualityComparer<TexlFunction>
+    {
+        public bool Equals(TexlFunction x, TexlFunction y)
+        {
+            return x.QualifiedName == y.QualifiedName && x.ParamTypes.SequenceEqual(y.ParamTypes);
```

Comment:
>ParamTypes [](http://example.com/codeflow?start=59&length=10)

This is actually a tricky function. 

ParamTypes may not be useful (hence need for "TypeRules"). 

We currently just use object reference equality directly.